6.0.0 MVU upgrade docs update (#946)
sourcegraph/docs: docs/admin/how-to/upgrade-postgres-12-16-builtin-dbs.mdx

# Upgrading Built-in PostgreSQL

The following doc contains detailed instructions for upgrading the built-in PostgreSQL databases. Via our `postgresql-16` and `postgresql-16-codeinsights` image entrypoint script. This doc assumes an admin is attempting to upgrade to Sourcegraph `6.0.0` from an older version (usually pre `5.10.0`) using one of our "deploy" repos. For more general info see [Upgrading PostgreSQL](/admin/postgres#upgrading-postgresql).

> > WARNING: Upgrading the PostgreSQL database requires stopping your Sourcegraph deployment which will result in **downtime**.

Reviewer: Warning calllout

# Upgrading Built-in PostgreSQL

The following doc contains detailed instructions for upgrading the built-in PostgreSQL databases. Via our `postgresql-16` and `postgresql-16-codeinsights` image entrypoint script. This doc assumes an admin is attempting to upgrade to Sourcegraph `6.0.0` from an older version (usually pre `5.10.0`) using one of our "deploy" repos. For more general info see [Upgrading PostgreSQL](/admin/postgres#upgrading-postgresql).

> WARNING: Upgrading the PostgreSQL database requires stopping your Sourcegraph deployment which will result in **downtime**.
>
> Additionally, once the upgrade process is started via the database container, interrupting the container before the upgrade is complete could result in corrupting the underlying Postgres database. **We strongly advise taking a backup before the upgrade.**

## Docker Compose

> Docker compose supports [standard](/admin/deploy/docker-compose/upgrade#standard-upgrades) and [multi-version](/admin/deploy/docker-compose/upgrade#multi-version-upgrades) upgrades to Postgres 16 versions! However if you prefer to upgrade your DBs independent of a Sourcegraph version upgrade see the following procedure.

Reviewer 2: ```suggestion
Docker compose supports [standard](/admin/deploy/docker-compose/upgrade#standard-upgrades) and [multi-version](/admin/deploy/docker-compose/upgrade#multi-version-upgrades) upgrades to Postgres 16 versions! However, if you prefer to upgrade your DBs independent of a Sourcegraph version upgrade see the following procedure.
```

1. **Bring down your containers**
```bash
docker-compose down --remove-orphans
```
2. Change the `image:` in your `docker-compose.yaml` to the release of `pgsql`, `codeinsights-db`, and `codeintel-db`. Or [merge in changes](/admin/deploy/docker-compose/upgrade#standard-upgrades) from the tagged sourcegraph release you're planning to upgrade to. **Example:**
```yaml
pgsql:
    container_name: pgsql
    image: 'index.docker.io/sourcegraph/postgresql-16:6.0.0'
...
codeintel-db:
    container_name: codeintel-db
    image: 'index.docker.io/sourcegraph/postgresql-16:6.0.0'
...
codeinsights-db:
    container_name: codeintel-db
    image: 'index.docker.io/sourcegraph/postgresql-16-codeinsights:6.0.0'
```
3. Bring up your deployments
```bash
docker-compose up -d pgsql codeintel-db codeinsights-db
```
Wait for the database containers to come up healthy. If for some reason the database containers fail to come up healthy, please check their logs and reach out to us for support at support@sourcegraph.com.

## Kubernetes Kustomize

> Kubernetes Kustomize supports [standard](/admin/deploy/kubernetes/upgrade#standard-upgrades) and [multi-version](/deploy/kubernetes/upgrade#multi-version-upgrades) upgrades to Postgres 16 versions! However if you prefer to upgrade your DBs independent of a Sourcegraph version upgrade see the following procedure.

Reviewer 2: ```suggestion
Kubernetes Kustomize supports [standard](/admin/deploy/kubernetes/upgrade#standard-upgrades) and [multi-version](/deploy/kubernetes/upgrade#multi-version-upgrades) upgrades to Postgres 16 versions! However, if you prefer to upgrade your DBs independent of a Sourcegraph version upgrade see the following procedure.
```

  **Example:**
  ```yaml
  # deploy-sourcegraph-k8s/base/sourcegraph/codeinsights-db/codeinsights-db.StatefulSet.yaml 
      containers:
        - name: codeinsights
          image: index.docker.io/sourcegraph/postgresql-16-codeinsights:6.0.0
  # deploy-sourcegraph-k8s/base/sourcegraph/codeintel-db/codeintel-db.StatefulSet.yaml 
      containers:
        - name: pgsql
          image: index.docker.io/sourcegraph/postgresql-16:6.0.0
  # deploy-sourcegraph-k8s/base/sourcegraph/pgsql/pgsql.StatefulSet.yaml 
      containers:
        - name: pgsql
          image: index.docker.io/sourcegraph/postgresql-16:6.0.0
  ```

3. Apply the new container images to the database statefulsets:
```sh
$ kubectl apply -f deploy-sourcegraph-k8s/base/sourcegraph/pgsql/pgsql.StatefulSet.yaml deploy-sourcegraph-k8s/base/sourcegraph/codeintel-db/codeintel-db.StatefulSet.yaml deploy-sourcegraph-k8s/base/sourcegraph/codeinsights-db/codeinsights-db.StatefulSet.yaml
```
- Wait until the containers come up, and complete their migration to Postgresql 16

## Helm

> To upgrade builtin containers in Helm admins must update their instance to v5.10.0 or v5.11.0 via the [helm multi-version upgrade procedure](/admin/deploy/kubernetes#multi-version-upgrade-procedure) or standard upgrade procedure. This is because `helm upgrade` doesn't support just the DBs in a chart.

Reviewer 2: ```suggestion
To upgrade builtin containers in Helm admins must update their instance to v5.11.6271 via the [helm multi-version upgrade procedure](/admin/deploy/kubernetes#multi-version-upgrade-procedure) or standard upgrade procedure. This is because `helm upgrade` doesn't support just the DBs in a chart.
```
